jwt/jwt.go

```diff
@@ -12,15 +12,32 @@ import (
 	"github.com/go-jose/go-jose/v3/jwt"
 )
 
+// AuthorizedPartyHandler is a type that can be used to perform checks
+// on the 'azp' claim.
 type AuthorizedPartyHandler func(string) bool
 
+// CustomClaimsConstructor can initialize structs for holding custom
+// JWT claims.
+type CustomClaimsConstructor func() any
+
 type VerifyParams struct {
 	// Token is the JWT that will be verified. Required.
 	Token string
 	// JWK the custom JSON Web Key that will be used to verify the
 	// Token with. Required.
-	JWK          *clerk.JSONWebKey
-	CustomClaims any
+	JWK *clerk.JSONWebKey
+	// CustomClaimsConstructor will be called when parsing the Token's
+	// claims. It's useful for parsing custom claims into user-defined
+	// types.
+	// Make sure it returns a pointer to a type (struct) that describes
```

Comment:
On second thought, it might be clearer to always request a pointer, just trying to think how the failure is manifested if a value is returned.

Reply:
When we parse the claims, we set the fields of the struct in a similar way that `UnmarshalJSON` operates. 

We could make a pointer out of a non-pointer, but it's hard to detect that.

I believe that we're better off always requesting a pointer. It's something that constructors usually return so it is expected.

There is no guarantee since the return type is `any`, but at least we can provide a guideline.